libs/sourcify/src/types.rs

```diff
@@ -0,0 +1,145 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum MatchType {
+    Full,
+    Partial,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct Error {
```

Comment:
maybe rename this to ErrorResponse ? I don't like idea of reimporting it as InternalError in `client.rs`